extension/fts/fts_indexing.cpp

```diff
@@ -0,0 +1,311 @@
+            ),
+            qtermids AS (
+                SELECT termid
+                FROM %fts_schema%.dict AS dict,
+                     tokens
+                WHERE dict.term = tokens.t
+            ),
+            qterms AS (
+                SELECT termid,
+                       docid
+                FROM %fts_schema%.terms AS terms
+                WHERE CASE WHEN fields IS NULL THEN 1 ELSE fieldid IN (SELECT * FROM fieldids) END
+                  AND termid IN (SELECT qtermids.termid FROM qtermids)
+            ),
+            subscores AS (
+                SELECT docs.docid,
+                       len,
+                       term_tf.termid,
+                       tf,
+                       df,
+                       (log(((SELECT num_docs FROM %fts_schema%.stats) - df + 0.5) / (df + 0.5))* ((tf * (k + 1)/(tf + k * (1 - b + b * (len / (SELECT avgdl FROM %fts_schema%.stats))))))) AS subscore
+                FROM (
+                    SELECT termid,
+                           docid,
+                           COUNT(*) AS tf
+                    FROM qterms
+                    GROUP BY docid,
+                             termid
+                ) AS term_tf
+                JOIN (
+                    SELECT docid
+                    FROM qterms
+                    GROUP BY docid
+                    HAVING CASE WHEN conjunctive THEN COUNT(DISTINCT termid) = (SELECT COUNT(*) FROM tokens) ELSE 1 END
+                ) AS cdocs
+                ON term_tf.docid = cdocs.docid
+                JOIN %fts_schema%.docs AS docs
+                ON term_tf.docid = docs.docid
+                JOIN %fts_schema%.dict AS dict
+                ON term_tf.termid = dict.termid
+            )
+            SELECT score
+            FROM (
+                SELECT docid,
+                       sum(subscore) AS score
+                FROM subscores
+                GROUP BY docid
+            ) AS scores
+            JOIN %fts_schema%.docs AS docs
+            ON  scores.docid = docs.docid
+            AND docs.name = docname
+        );
+    );
+
+    // we may have more than 1 input field, therefore we union over the fields, retaining information which field it came from
+	string tokenize_field_query = SQL(
+        SELECT unnest(%fts_schema%.tokenize(fts_ii."%input_value%")) AS w,
+	           rowid AS docid,
+	           (SELECT fieldid FROM %fts_schema%.fields WHERE field = '%input_value%') AS fieldid
+        FROM %input_schema%.%input_table% AS fts_ii
+    );
+	vector<string> field_values;
+	vector<string> tokenize_fields;
+	for (idx_t i = 0; i < input_values.size(); i++) {
+		field_values.push_back(StringUtil::Format("(%i, '%s')", i, input_values[i]));
+        tokenize_fields.push_back(StringUtil::Replace(tokenize_field_query, "%input_value%", input_values[i]));
+	}
+    result = StringUtil::Replace(result, "%field_values%", StringUtil::Join(field_values, ", "));
```

Comment:
The formatting of this file seems off. I think you added it to the ignore list of the formatter because of the `SQL` blocks, perhaps instead annotate those with `// clang-format off` and `// clang-format on` [as described here](https://stackoverflow.com/questions/33258159/how-to-make-clang-format-skip-sections-of-c-code)?

Reply:
Much cleaner solution to the problem. Done.